src/main/java/org/opentripplanner/middleware/bugsnag/jobs/BugsnagEventHandlingJob.java

```diff
@@ -71,20 +78,36 @@ private void refreshEventRequest(BugsnagEventRequest request) {
             LOG.error("Failed to refresh event request");
             return;
         }
-        if (!refreshedRequest.status.equalsIgnoreCase("completed")) {
-            // Request not completed by Bugsnag yet. Return and await the next cycle/refresh.
-            // TODO: Update the request data in the database? What if the status has changed since the last fetch?
-            return;
-        }
-        // First, remove "stale" requests (i.e., those that are older than this latest one).
-        Persistence.bugsnagEventRequests.removeFiltered(Filters.lte("dateCreated", request.dateCreated));
-        // Next, get and store the new events from the completed request and notify users.
-        List<BugsnagEvent> newEvents = getNewEvents(refreshedRequest);
-        if (newEvents.size() > 0) {
-            LOG.info("Found {} new events. Storing and notifying subscribed admin users.", newEvents.size());
-            Persistence.bugsnagEvents.createMany(newEvents);
-            // Notify any subscribed users about new events.
-            sendEmailForEvents(newEvents);
+        switch (refreshedRequest.status.toLowerCase()) {
+            case "completed":
+                // First, delete the last completed request for the project.
+                BugsnagEventRequest latestCompletedRequestForProject =
+                    BugsnagJobs.getLatestCompletedRequestForProject(request.projectId);
+                latestCompletedRequestForProject.delete();
+                // Next, replace the newly completed request.
+                request.update(refreshedRequest);
+                // Finally, get and store the new events from the completed request and notify users.
+                List<BugsnagEvent> newEvents = getNewEvents(refreshedRequest);
+                if (newEvents.size() > 0) {
+                    LOG.info("Found {} new events. Storing and notifying subscribed admin users.", newEvents.size());
+                    Persistence.bugsnagEvents.createMany(newEvents);
+                    // Notify any subscribed users about new events.
+                    sendEmailForEvents(newEvents.size());
+                }
+                break;
+            case "expired":
+                // First, remove the expired request.
+                LOG.info("Event data request for project {} has expired. Removing from database.", request.projectId);
+                request.delete();
+                // Next, immediately trigger a new event data request to replace the expired one.
+                LOG.info("Triggering a new event data request for project {} to replace previously expired.", request.projectId);
+                BugsnagEventRequestJob.triggerEventDataRequestForProject(request.projectId, request.daysInPast);
```

Comment:
Shouldn't this recalculate the `daysInPast` variable in case an expired request has occurred multiple days in the past?